src/use-swr.ts

```diff
@@ -276,14 +276,25 @@ function useSWR<Data = any, Error = any>(
     fn = config.fetcher
   }
 
+  const willRevalidateOnMount = () => {
+    return (
+      config.revalidateOnMount ||
+      (!config.initialData && config.revalidateOnMount === undefined)
+    )
+  }
+
   const resolveData = () => {
     const cachedData = cache.get(key)
     return typeof cachedData === 'undefined' ? config.initialData : cachedData
   }
 
+  const resolveIsValidating = () => {
+    return !!cache.get(keyValidating) || willRevalidateOnMount()
```

Comment:
```suggestion
    return !!cache.get(keyValidating) || (key && willRevalidateOnMount())
```

We need to also check `key` here because if it's falsy, the initial revalidation won't be triggered.